### DISCUSSION TITLE
expect_reply versus ACK

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1407] [General]
The request() API relies on expect_reply to determine whether to set TransmitOption.ACK.

[Source Code](https://github.com/zigpy/zigpy/blob/150e88c774c062e8321822437bc7101ca26f7dff/zigpy/application.py#L838)

The issue is that when I perform a read_attribute_request to a device behind a router, the request() API responds with SUCCESS as soon as the request is sent to the next hop.

I believe we should decouple expect_reply (which can be a response like a read attribute response or a default response) from TransmitOption.ACK, which ensures the message is received by the targeted device.

**Comment by puddly** [comment]
The `request` API is deprecated and, as you can see in the source code, just a thin wrapper around `send_packet`. I suggest you just use `send_packet` directly.
